src/foam/nanos/theme/Theme.js

```diff
@@ -584,5 +592,131 @@ foam.CLASS({
         return foam.util.SafetyUtil.isEmpty(getName()) || foam.util.SafetyUtil.isEmpty(getDescription()) ? "" : getName() + " " + getDescription();
       `
     },
+    {
+      name: 'merge',
+      type: 'Theme',
+      args: [ 'Theme other' ],
+      code: function(other) {
+        var theme = this.clone();
+        var props = this.cls_.getAxiomsByClass(foam.core.Property);
+        for ( var i = 0 ; i < props.length ; i++ ) {
+          var name = props[i].name;
+          if ( ! other.hasOwnProperty(name) ) continue;
+
+          if ( foam.core.StringArray.isInstance(props[i])          ) this.mergeArrayProperty(props[i], theme, other);
+          else if ( foam.core.Map.isInstance(props[i])             ) this.mergeMapProperty(props[i], theme, other);
+          else if ( foam.core.FObjectProperty.isInstance(props[i]) ) this.mergeFObjectProperty(props[i], theme, other);
+          else
+            theme[name] = other[name];
+        }
+        return theme;
+      },
+      javaCode: `
+        var theme = (Theme) this.fclone();
+        List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
+        for ( PropertyInfo p : props ) {
+          if ( ! p.isSet(other) ) continue;
+
+          if ( p.getValueClass().isArray()                            ) mergeArrayProperty(p, theme, other);
+          else if ( Map.class.isAssignableFrom(p.getValueClass())     ) mergeMapProperty(p, theme, other);
+          else if ( FObject.class.isAssignableFrom(p.getValueClass()) ) mergeFObjectProperty(p, theme, other);
+          else
+            p.set(theme, p.get(other));
+        }
+        return theme;
+      `
+    },
+    {
+      name: 'mergeArrayProperty',
+      args: [ 'PropertyInfo prop', 'Theme t1', 'Theme t2' ],
+      code: function(prop, t1, t2) {
+        var name = prop.name;
+
+        if ( ! t1.hasOwnProperty(name) ) t1[name] = t2[name];
+        else if ( foam.util.equals(t1[name], t2[name]) ) return;
```

Comment:
I think the code would be clearer if this where if ( ! foam.util.equals(t1[name], t2[name]) ) {
  code below here.
}

Reducing from a 3 clause if/else statement to just 2. Same pattern replicated multiple times for both JS and java.

Reply:
Fixed in https://github.com/kgrgreer/foam3/pull/520/commits/051c1ed972ac386fe536224a65d1e73bd2482d57.